### ISSUE TITLE
Unable to display agents [Datahandler factory. Error. 3013]

### ISSUE THREAD

**Issue by fayax** [wazuh/wazuh#853]
I'm unable to see the agents from the agents tab.
I get this error message: Datahandler factory. Error. 3013 - Maximum attempts exceeded for sqlite3 execute: database is locked
Any suggestions?

Thanks

**Comment by JuanjiJG** [comment]
Hello @fayax,

That `sqlite3` error could mean that there have been multiple executions at the same time, and the database got blocked by some process.

Could you please tell us what's your current version of Wazuh manager?

Regards,
Juanjo

**Comment by fayax** [comment]
Hi @JuanjiJG 

I have `Wazuh v3.3.1` currently installed.

**Comment by JuanjiJG** [comment]
Hello again @fayax,

Let's try something. Restart your manager with the following command:
```
systemctl restart wazuh-manager
```
This way, the database will be restarted and all the possible executions will be stopped. Now, we're going to execute an API request to see if it's capable of returning the list of agents:
```
curl -u <API_USER>:<API_PASSWORD> http(s)://<API_URL>:55000/agents?pretty
```
Let me know if you get a proper response from the API (that is, the list of agents).

Thanks for your patience.

Regards,
Juanjo

**Comment by fayax** [comment]
Hi @JuanjiJG 

I was able to get the full list of agents via curl. 
No issues at the moment after restarting.
So some process is blocking database after some time?

Thanks,

**Comment by JuanjiJG** [comment]
Hello again @fayax,

I'm glad everything got solved.

Definitely, this is not a usual behaviour. We use sqlite3 as the database engine, and sometimes it can be slow and get stuck in heavy environments with a large number of agents. We have future plans for improving the database system, but for now, I can't give you specific details or ETAs about it.

We're sorry for this inconveniences. Don't hesitate to open a new issue if you have more questions or problems.

I'll proceed to close this issue.

Regards,
Juanjo
